### DISCUSSION TITLE
Can I use wrapper function for `useShallow`?

### DISCUSSION THREAD

**Discussion by reconcrete** [pmndrs/zustand#2828] [Q&A] [answered]
Can I do something like that:

```typescript
import { create } from 'zustand'

const useStore = create((set) => ({
  count: 1,
  inc: () => set((state) => ({ count: state.count + 1 })),
}))

const useStoreWithShallow =  <idk what to put here>

function Counter() {
  const [ count, inc ] = useStoreWithShallow((state) => [state.count, state.inc]))
  return (
    <div>
      <span>{count}</span>
      <button onClick={inc}>one up</button>
    </div>
  )
}

```

**Comment by dbritto-dev** [comment, marked as answer]
@Yoha485 here you go -> https://zustand.docs.pmnd.rs/apis/create-with-equality-fn

**Reply by reconcrete** [reply]
Thanks! That will help
To go even deeper, can I ask another question:

As far as I understand if you create store with just `create` function, your previous and new states would be compared with just `===` operator. And if I create store with `createWithEqualityFn` and pass simply `shallow` to that, it will compare my states with "Object.is" or something like that

But if I want to use both `===` and `Object.is` in different places, can there still be a wrapper function declaration like in my original question? Will it work fine?

**Reply by dbritto-dev** [reply]
@Yoha485 not with `===` but `Object.is` instead

**Reply by dbritto-dev** [reply]
@Yoha485 `shallow` do better than `Object.is`, you can find what `shallow` can do here -> https://github.com/pmndrs/zustand/blob/main/tests/vanilla/shallow.test.tsx

**Reply by dbritto-dev** [reply]
but in a nutshell, you can compare  `Generator`s, `Array`s, `Map`s, `Set`s, `Object`s and `Primitive`s
